src/main/java/uk/gov/digital/ho/hocs/search/application/aws/SearchListener.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.digital.ho.hocs.search.application.aws;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.cloud.aws.messaging.listener.SqsMessageDeletionPolicy;
+import org.springframework.cloud.aws.messaging.listener.annotation.SqsListener;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.search.api.CaseDataService;
+import uk.gov.digital.ho.hocs.search.api.dto.*;
+import uk.gov.digital.ho.hocs.search.application.LogEvent;
+import uk.gov.digital.ho.hocs.search.application.queue.CreateAuditDto;
+import uk.gov.digital.ho.hocs.search.application.queue.EventType;
+import uk.gov.digital.ho.hocs.search.domain.exceptions.ApplicationExceptions;
+
+@Service
+public class SearchListener {
+
+    private final ObjectMapper objectMapper;
+    private final CaseDataService caseDataService;
+
+    public SearchListener(ObjectMapper objectMapper,
+                          CaseDataService caseDataService) {
+        this.objectMapper = objectMapper;
+        this.caseDataService = caseDataService;
+    }
+
+    @SqsListener(value = "${aws.sqs.search.url}", deletionPolicy = SqsMessageDeletionPolicy.ON_SUCCESS)
+    public void onSearchEvent(String message) throws JsonProcessingException {
+        CreateAuditDto request = objectMapper.readValue(message, CreateAuditDto.class);
+
+        EventType type = EventType.fromString(request.getType());
+
+        if(type != null) {
+            switch (type) {
+                case CASE_CREATED:
```

Comment:
I'd be tempted to see if we could abstract this somehow..

Reply:
going to address with another ticket https://collaboration.homeoffice.gov.uk/jira/browse/HOCS-4007